### PR TITLE
Removes sleep() call from clockwork and floor turfs

### PR DESCRIPTION
**Pull request by chuga-git**
## What Does This PR Do

[This PR ](https://github.com/ParadiseSS13/Paradise/pull/24209) added a delayed effect to walking over clockwork and clock turfs, which used sleep for the delay. This turns the sleep into an addtimer callback because sleeping on Entered makes the linter go haywire and has the potential to break signals. 

Also converts revenant death animation to use animate() instead of decrementing alpha in a for loop in between sleeps. Fantastic.

## Why It's Good For The Game

It is making my code not compile and I would like it to compile.

## Testing

Made sure it does the same thing and doesn't scream bloody murder.

### CODE REVIEW

File: code/game/turfs/simulated/floor/plating.dm [hunk 1]
```diff
@@ -246,8 +246,11 @@
 			return
 
 	if(!. && isliving(A))
-		sleep(2 DECISECONDS)
-		new /obj/effect/temp_visual/cult/turf/open/floor(src)
+		addtimer(CALLBACK(src, PROC_REF(spawn_visual)), 0.2 SECONDS, TIMER_DELETE_ME)
+
+
```
Comment (1080pCat) [inline code/game/turfs/simulated/floor/plating.dm:251]:
```suggestion

```

File: code/game/gamemodes/miniantags/revenant/revenant.dm [hunk 2]
```diff
@@ -223,9 +223,7 @@
 	playsound(src, 'sound/effects/screech.ogg', 100, 1)
 	visible_message("<span class='warning'>[src] lets out a waning screech as violet mist swirls around its dissolving body!</span>")
 	icon_state = "revenant_draining"
-	for(var/i = alpha, i > 0, i -= 10)
-		sleep(0.1)
-		alpha = i
+	animate(src, alpha = 0, time = 3 SECONDS)
```
Comment (Contrabang) [inline code/game/gamemodes/miniantags/revenant/revenant.dm:226]: animate is async, unlike this previous sleep call. Can you make sure that revs still die correctly with this?

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/miniantags/revenant/revenant.dm (hunk 2 above)
Comment (chuga-git) [reply code/game/gamemodes/miniantags/revenant/revenant.dm:226]: I tested it on a clientless revenant by calling the proc directly and then setting its damage high enough to kill it. In both cases, it died successfully.

**Review by Contrabang — APPROVED** [review]

**Review by DGamerL — APPROVED** [review]
I love being confronted with my own shitcode 👍

**Pull request merged**